File: java/src/test/java/rocks/iulian/cnp/RomanianPersonalNumberTest.java
package rocks.iulian.cnp;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.hamcrest.CoreMatchers.instanceOf;

public class RomanianPersonalNumberTest {

    @Test
    public void shouldCreateCnp() {
        RomanianPersonalNumber number = RomanianPersonalNumber.fromString("1690509049993");
        assertThat(number, instanceOf(Cnp.class));
    }

    @Test
    public void shouldCreateNif() {
        RomanianPersonalNumber number = RomanianPersonalNumber.fromString("9000123456785");
        assertThat(number, instanceOf(Nif.class));
    }

    @Test
    public void shouldBeValid() {
        assertTrue(RomanianPersonalNumber.isValid("1690509049993"));
    }

    @Test
    public void shouldBeInvalid() {
        assertFalse(RomanianPersonalNumber.isValid("1690509049990"));
    }
}

File: ruby/test/romanian_personal_number_test.rb
require 'test/unit'
require_relative '../lib/cnp'

class RomanianPersonalNumberTest < Test::Unit::TestCase
  def test_create_cnp
    number = Cnp::RomanianPersonalNumber.from_string("1690509049993")
    assert number.kind_of?(Cnp::Cnp)
  end

  def test_create_nif
    number = Cnp::RomanianPersonalNumber.from_string("9000123456785")
    assert number.kind_of?(Cnp::Nif)
  end

  def test_valid
    assert Cnp::RomanianPersonalNumber.valid?("1690509049993")
  end

  def test_invalid
    assert_false Cnp::RomanianPersonalNumber.valid?("1690509049990")
  end
end